include/anyrpc/xml/xmlwriter.h

```diff
@@ -42,8 +42,8 @@ class ANYRPC_API XmlWriter : public Handler
     //! Make the text more readable with indentation and linefeeds
     void SetPretty(bool pretty=true) { pretty_ = pretty; }
     //! Set the double format method. The default is no exponents for xmlrpc spec compatibility.
-    void SetScientificPrecision(unsigned precision=18)
-        { precision_ = std::min( 32u, precision_); }
+    void SetScientificPrecision(unsigned precision=32)
```

Comment:
Here we have two things, first the patch ensures that the precision gets actually set (unref'd parameter warnings helped to spot it).

Changing the default is merely for consistency, as the precision gets limited to 32 anyway. More a matter of taste...